### ISSUE TITLE
Replace `read_only` parameter with a list of HTTP methods

### ISSUE THREAD

**Issue by dantownsend** [piccolo-orm/piccolo_api#121]
`PiccoloCRUD` has an option called `read_only`.

I think it's worth phasing this out in favour of an argument called `methods`.

It could be something like:

```python
PiccoloCRUD(methods=[HTTP.get])
```

If `read_only` is provides in kwargs, then we can just map that to `methods=[HTTP.get]` and provide a deprecation warning.

It means you have more fine grained control over which methods to expose:

```python
PiccoloCRUD(methods=[HTTP.get, HTTP.post])
```

**Comment by sinisaos** [comment]
@dantownsend I don't think we should change this because `read_only` is a good, safe and simple mechanism (if we pass `False` to `read_only` only the `GET` method is allowed, otherwise all methods are allowed) to show or hide certain endpoints. The documentation around that is also good. Sorry if I miss the point.

**Comment by dantownsend** [comment]
@sinisaos I can imagine use cases where you just want to use part of `PiccoloCRUD`. At the moment, there's just the option for `GET`, or everything.

I agree that `read_only` is very convenient and easy to understand, but it doesn't support the middle ground where maybe you want `GET` and `POST,` but not `PATCH`, `PUT` and `DELETE`.

I don't think it's high priority, but maybe let's keep it open, and see if anybody else encounters this issue, in which case we can prioritise it. Or there might even be a better solution to just exposing certain methods.

**Comment by sinisaos** [comment]
@dantownsend You're right. Probably most users want to use all `CRUD` methods, but yes, there may be some edge cases where some users only wants certain methods. Sorry, I didn't think of this that way. Thank you for the explanation. Cheers.
